crates/containerd-shim-wasmtime/src/instance.rs

```diff
@@ -79,6 +79,15 @@ impl<T: WasiConfig> Default for WasmtimeEngine<T> {
     fn default() -> Self {
         let mut config = T::new_config();
         config.async_support(true); // must be on
+
+        if let Some(true) = use_pooling_allocator_by_default()
+            .context("failed to determine allocation strategy")
+            .unwrap()
```

Comment:
**Fix:** I think we should fail back to the slow behaviour instead of panic-ing:
```suggestion
            .unwrap_or(None)
```